pointer-masking-proposal.adoc

```diff
@@ -295,76 +298,133 @@ MASK
+,===
+hpmbase[31:0]
+BASE
+,===
+Figure 7b: Memory Tagging Extension register (_**hpmbase**_) for RV32 (H-MODE) +
+ +
+
 [%header, cols=1*]
 ,===
 spmbase[31:0]
 BASE
 ,===
-Figure 7a: Memory Tagging Extension register (_**spmbase**_) for RV32 (S-MODE) +
+Figure 7c: Memory Tagging Extension register (_**spmbase**_) for RV32 (S-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 upmbase[31:0]
 BASE
 ,===
-Figure 7b: Memory Tagging Extension register (_**upmbase**_) for RV32 (U-MODE) +
+Figure 7d: Memory Tagging Extension register (_**upmbase**_) for RV32 (U-MODE) +
+ +
+
+[%header, cols=1*]
+,===
+mpmbase[63:0]
+BASE
+,===
+Figure 8a: Memory Tagging Extension register (_**mpmbase**_) for RV64 (M-MODE) +
+ +
+
+[%header, cols=1*]
+,===
+hpmbase[63:0]
+BASE
+,===
+Figure 8b: Memory Tagging Extension register (_**hpmbase**_) for RV64 (H-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 spmbase[63:0]
 BASE
 ,===
-Figure 8a: Memory Tagging Extension register (_**spmbase**_) for RV64 (S-MODE) +
+Figure 8c: Memory Tagging Extension register (_**spmbase**_) for RV64 (S-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 upmbase[63:0]
 BASE
 ,===
-Figure 8b: Memory Tagging Extension register (_**upmbase**_) for RV64 (U-MODE) +
+Figure 8d: Memory Tagging Extension register (_**upmbase**_) for RV64 (U-MODE) +
+ +
+
+[%header, cols=1*]
+,===
+mpmbase[127:0]
+BASE
+,===
+Figure 9a: Memory Tagging Extension register (_**mpmbase**_) for RV128 (M-MODE) +
+ +
+
+[%header, cols=1*]
+,===
+hpmbase[127:0]
+BASE
+,===
+Figure 9b: Memory Tagging Extension register (_**hpmbase**_) for RV128 (H-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 spmbase[127:0]
 BASE
 ,===
-Figure 9a: Memory Tagging Extension register (_**spmbase**_) for RV128 (S-MODE) +
+Figure 9c: Memory Tagging Extension register (_**spmbase**_) for RV128 (S-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 upmbase[127:0]
 BASE
 ,===
-Figure 9b: Memory Tagging Extension register (_**upmbase**_) for RV128 (U-MODE) +
+Figure 9d: Memory Tagging Extension register (_**upmbase**_) for RV128 (U-MODE) +
  +
 
 [[explanation]]
 ## Explanation
 
-_**MMTE**_ register fully two-fold function:
+_**MMTE**_ register fulfills two-fold function:
 
-1. Can be programmable by more privileged mode
+1. Can be programmable by more privileged mode (unless PM.Current bit is enabled)
 2. Performs status register function for the current privilege mode
 
-_PM_ bits from **_MMTE_** register are accessible in all modes (U/S/HS/M) and can be read to query the current status of PM feature (if enabled). However, only M-mode privileged code can set the value for _PM_ bits.
+_PM_ bits from **_MMTE_** register are accessible in all modes (U/S/HS/M) and can be read to query the current status of PM feature (if enabled). By default, only higher privileged code can set the value for _PM_ bits. However, higher privileged code can enable _PM.Current_ bit for lower privileged code. In such scenario, current privilege code has a possibility to self-manage its own configuration of _PM_ bits.
```

Comment:
Some sentences in text imply that PM bits for each privilege level exist, while register layout has nothing like that.

Reply:
Correct. There are separate PM bits per each level.

Reply:
So if we have 4 bits per each level it seems a little bit strange that we can't read or set them directly.

Consider the following situations:

1. Suppose we are in S mode. What is the procedure to figure out the value of **PM.Enable** for U mode? For example - consider a scenario when U-mode and S-mode SW have an ability to self-manage the state of the pointer-masking subsystem?
2. What is the procedure to enable pointer-masking from S mode? To me it seems that while in S-mode we must set PM. Enable bit to 1 and then perform a transition to U mode. But this also that we have to enable PointerMasking functionality for S mode too! This looks strange. 

Could you please clarify?